pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -227,6 +225,19 @@ func (r *KsvcReconciler) Reconcile() (*knservingv1.ServiceStatus, error) {
 		}
 		return nil, err
 	}
+	// Set ResourceVersion which is required for update operation.
+	desired.ResourceVersion = existing.ResourceVersion
+	// Add immutable annotations to avoid validation error during dry-run update.
+	desired.Annotations[knserving.CreatorAnnotation] = existing.Annotations[knserving.CreatorAnnotation]
```

Comment:
Is updatorAnnotation immutable ? The updator for desired service can be different from existing service.

Reply:
No updatorAnnotation is not immutable. Only the creatorAnnotation is immutable.

Reply:
Right, my question is that the desired `updatedAnnotation` can be different from existing one, so we should not set the desired updater annotation to the creator annotation right?

Reply:
Discussed with @sivanantha321 Knative has the check to prevent user setting the `updatorAnnotation`.